### PR TITLE
Fix tickets list from assets

### LINKED ISSUE #2151: [9.1.3] Bug: you can see the open tickets in the tab of the tickets associated with the item
If you go to an item and select the tickets tab you can see the last 6 tickets opened + the number of tickets associated.

### PR DESCRIPTION
**Pull request by trasher**
Display incoming tickets if user can assign
Do not rely on groups if list is empty

Fixes #2151

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2151 #2078

### CODE REVIEW

File: inc/ticket.class.php [hunk 1]
```diff
@@ -5284,23 +5284,32 @@ static function showListForItem(CommonDBTM $item) {
                         " AND `glpi_items_tickets`.`itemtype` = '".$item->getType()."')";
 
 
+            $restrict_with_rights = "";
             // you can only see your tickets
             if (!Session::haveRight(self::$rightname, self::READALL)) {
-               $restrict .= " AND (`glpi_tickets`.`users_id_recipient` = '".Session::getLoginUserID()."'
+               $restrict_with_rights .= " AND (`glpi_tickets`.`users_id_recipient` = '".Session::getLoginUserID()."'
```
Comment (orthagh) [inline inc/ticket.class.php]: so remove AND here

**Comment by yllen** [comment]
Not agree with all PR.
For this problem you only have to change the request construct to see Ticket in incoming status if item is link to ticket

` default :
            $restrict = "(`glpi_items_tickets`.`items_id` = '".$item->getID()."' ".
                        " AND `glpi_items_tickets`.`itemtype` = '".$item->getType()."'";

            // in list of ticket linked to the object, i also display incoming ticket if i have correct right
            if (Session::haveRightsAnd(self::$rightname, [self::READASSIGN, self::ASSIGN])) {
               $restrict .= " AND glpi_tickets.status = ".self::INCOMING;
            }
            $restrict .= ")";


            // you can only see your tickets
     `

**Comment by trasher** [comment]
I'm not sure I understand... « in list of ticket linked to the object »; but if we are in the `showListForItem` method, which is only called to list linked tickets from an object; aren't we?

**Comment by trasher** [comment]
Will be fixed another way.

**Pull request closed without merging**